pkg/bootstrapper/bootstrapper.go

```diff
@@ -499,63 +502,99 @@ func (wmcb *winNodeBootstrapper) createKubeletService() error {
+
+	if wmcb.kubeletSVC == nil {
+		if err := wmcb.createKubeletService(c, kubeletArgs); err != nil {
+			return fmt.Errorf("failed to create kubelet service : %v ", err)
+		}
+	} else {
+		if err := wmcb.updateKubeletService(c, kubeletArgs); err != nil {
+			return fmt.Errorf("failed to update kubelet service : %v ", err)
+		}
+	}
+
+	if err := wmcb.kubeletSVC.setRecoveryActions(); err != nil {
+		return fmt.Errorf("failed to set recovery actions for Windows service %s : %v", KubeletServiceName, err)
+	}
+	return nil
+}
+
+// createKubeletService creates a new kubelet service to our specifications
+func (wmcb *winNodeBootstrapper) createKubeletService(c mgr.Config, kubeletArgs []string) error {
 	ksvc, err := wmcb.svcMgr.CreateService(KubeletServiceName, filepath.Join(wmcb.installDir, "kubelet.exe"), c, kubeletArgs...)
 	if err != nil {
 		return err
 	}
 
-	var dependents []*mgr.Service
-	if dependentSvc, err := wmcb.svcMgr.OpenService(kubeletDependentSvc); err == nil {
-		dependents = append(dependents, dependentSvc)
-	}
-	wmcb.kubeletSVC, err = newKubeletService(ksvc, dependents)
+	wmcb.kubeletSVC, err = newKubeletService(ksvc, nil)
 	if err != nil {
 		return fmt.Errorf("could not initialize struct kubeletService: %v", err)
 	}
+	return nil
+}
 
-	if err := wmcb.kubeletSVC.setRecoveryActions(); err != nil {
-		return fmt.Errorf("failed to set recovery actions for Windows service %s", KubeletServiceName)
+// updateKubeletService updates an existing kubelet service with our specifications
+func (wmcb *winNodeBootstrapper) updateKubeletService(config mgr.Config, kubeletArgs []string) error {
+	// Get existing config
+	existingConfig, err := wmcb.kubeletSVC.config()
+	if err != nil {
+		return fmt.Errorf("no existing config found")
 	}
 
-	return nil
-}
+	// Stop the kubelet service as there could be open file handles from kubelet.exe on the plugin files
+	if err := wmcb.kubeletSVC.stop(); err != nil {
+		return fmt.Errorf("unable to stop kubelet service: %v", err)
+	}
+	// Populate existing config with non default values from desired config.
+	existingConfig.Dependencies = config.Dependencies
+	existingConfig.DisplayName = config.DisplayName
+	existingConfig.StartType = config.StartType
 
-// refreshServiceManager will disconnect and reconnect from the Windows service API. In order to complete certain
-// operations, there must be zero handlers to the API present on the system.
-func (wmcb *winNodeBootstrapper) refreshServiceManager() error {
-	var err error
-	if err = wmcb.Disconnect(); err != nil {
-		return err
+	// Create kubelet command to populate config.BinaryPathName
+	// Add a space after kubelet.exe followed by the stand alone args
+	kubeletcmd := " " + filepath.Join(wmcb.installDir, "kubelet.exe") + " "
+	var standaloneArgs string
+	var keyvalueArgs string
+	// Add rest of the args
+	for _, args := range kubeletArgs {
+		if strings.Contains(args, "=") {
+			keyvalueArgs += args + " "
+		} else {
+			standaloneArgs += args + " "
+		}
```

Comment:
Why is this separation being done?

Reply:
Thanks for catching that. I need to revert it. Added it as part of a debug.